### ISSUE TITLE
Dataframe input does not refresh as intended

### ISSUE THREAD

**Issue by tnwei** [gradio-app/gradio#921]
### Describe the bug

When using a dataframe as input, the Gradio app output is locked to the input received when run for the first time. Subsequent user updates to the dataframe is not reflected in the Gradio app output. In addition, when "Clear" is selected to clear the outputs, the Gradio app behaves as if an empty dataframe is passed. The app has to be relaunched to have new input processed correctly.

Reporting both instances in the same issue as I suspect they both have to do with how Gradio handles dataframe input under the hood.

### Reproduction

Both of these are reproducible using the `filter_records.ipynb` demo at https://colab.research.google.com/drive/1mSrwzeNRMQc0FBiUY9Yx_rwqyX1X-7Cf#scrollTo=RN-8fjhplcnX.  Code reproduced below for convenience:

```
import gradio as gr

def filter_records(records, gender):
    return records[records["gender"] == gender]

iface = gr.Interface(
    filter_records,
    [
        gr.inputs.Dataframe(
            headers=["name", "age", "gender"],
            datatype=["str", "number", "str"],
            row_count=5,
        ),
        gr.inputs.Dropdown(["M", "F", "O"]),
    ],
    "dataframe",
    description="Enter gender as 'M', 'F', or 'O' for other.",
)

iface.test_launch()

iface.launch()
```

Including reproduction steps in the screenshot section below.

### Screenshot

1. Launch the app, key in the following table, and run the app. The output will correctly filter to show only the first row.

| name | age | gender |
|--------|-----|----------|
| A        | 1    | M          |
| B        | 1    | F          |
| C        | 1    | O          |

![image](https://user-images.githubusercontent.com/12769364/161469486-93582fbd-90b3-4c99-945c-ba9fb88dad9e.png)

2. Change `gender` in rows 2 and 3 to "M", and submit the new input. All rows should be displayed as expected output, but the output did not change.

![image](https://user-images.githubusercontent.com/12769364/161469558-f13559f9-5055-4c11-92be-d75223851253.png)

3. Select "Clear" then "Submit". The output is now blank, as if no rows are given as input.

![image](https://user-images.githubusercontent.com/12769364/161469598-257b9a51-5700-4de1-b718-b342725538b1.png)

### Logs

```shell
No logs generated
```

### System Info

```shell
Gradio version: 2.9.1, running on Google Colab
OS: MS Windows
Browser: Edge
```

### Severity

blocker

**Comment by Olvi73** [comment]
Same problem, if i change the data and doesnt click clear button,the dataframe will not refresh. Not only change the data manually but also use `Examples` to input, but when i click `Clear` button then `Submit`, the output will refresh.
My app's input is image, but yours are dataframe, when i click the `Clear`button ,it seems doesnt work at all, think the problem is input dataframe part

**Comment by abidlabs** [comment]
Thanks for letting us know @tnwei and @Olvi73, we are working on fixing all of the issues with the `Dataframe` component.

cc @pngwn

**Comment by pngwn** [comment]
Closed by #981.

**Comment by tnwei** [comment]
Awesome, thank you!
